src/blockMonitor/confirmationObserver.ts

```diff
@@ -1,17 +1,19 @@
 import { StartStopService, ArgumentError, BlockThresholdReachedError, ReorgError } from "../dataEntities";
 import { BlockProcessor } from "./blockProcessor";
 import { CancellablePromise, cancellablePromiseRace } from "../utils";
+import { HasTxHashes, IBlockStub } from "../dataEntities/block";
+import { getConfirmations } from "./blockCache";
 
 // A TransactionListener fulfills/rejects a promise if possible; returns true on success
 type TransactionListener = (blockNumber: number, blockHash: string) => boolean;
 
 /**
  * Allows to observe transactions to be notified when they reach a given number of confirmations.
  */
-export class ConfirmationObserver extends StartStopService {
+export class ConfirmationObserver<T extends IBlockStub & HasTxHashes> extends StartStopService {
     private txListeners = new Set<TransactionListener>();
 
-    constructor(private readonly blockProcessor: BlockProcessor) {
+    constructor(private readonly blockProcessor: BlockProcessor<T>) {
```

Comment:
I dont think you should choose T here - just IBlockStub & HasTxHashes, that's all that ConfirmationObserver needs to function.

Reply:
Done.